extensions/wikia/FeedsAndPosts/Discussion/DiscussionGateway.php

```diff
@@ -59,6 +59,32 @@ public function getPollVoters( string $pollId, ?string $answerId ) {
 		} );
 	}
 
+	public function upVotePost( string $postId, int $userId, array $userTraceHeaders ) {
+		return $this->makeCall( function () use ( $postId, $userId, $userTraceHeaders ) {
+			return $this->httpClient->post(
+				"{$this->serviceUrl}/internal/{$this->wikiId}/votes/post/{$postId}",
+				[
+					RequestOptions::HEADERS => [ WebRequest::WIKIA_INTERNAL_REQUEST_HEADER => '1',
+												 Constants::HELIOS_AUTH_HEADER => $userId ] +
+											   $userTraceHeaders,
+					RequestOptions::TIMEOUT => 3.0,
+				] );
+		} );
+	}
+
+	public function downVotePost( string $postId, int $userId, array $userTraceHeaders ) {
+		return $this->makeCall( function () use ( $postId, $userId, $userTraceHeaders ) {
+			return $this->httpClient->delete(
+				"{$this->serviceUrl}/internal/{$this->wikiId}/votes/post/{$postId}",
+				[
+					RequestOptions::HEADERS => [ WebRequest::WIKIA_INTERNAL_REQUEST_HEADER => '1',
+												 Constants::HELIOS_AUTH_HEADER => $userId ] +
+											   $userTraceHeaders,
+					RequestOptions::TIMEOUT => 3.0,
```

Comment:
I wonder if we should have this timeout set as a const + share it between all requests.

Reply:
yeah, I can use constant here